server/lbry/apis.go

```diff
@@ -91,6 +91,46 @@ func (c apiClient) CheckPerk(options CheckPerkOptions) (bool, error) {
 	return perkRes.Data.HasAccess, nil
 }
 
+type ArweavePaymentDetailsResponse struct {
+	Amount         uint64    `json:"amount"`
+	Currency       string    `json:"currency"`
+	Status         string    `json:"status"`
+	UserId         uint64    `json:"user_id"`
+	ChannelClaimId string    `json:"channel_claim_id"`
+	TippedAt       time.Time `json:"tipped_at"`
+}
+type apiArweavePaymentDetailsResponse struct {
+	Success bool                          `json:"success"`
+	Error   interface{}                   `json:"error"`
+	Data    ArweavePaymentDetailsResponse `json:"data"`
+}
+
+func (c apiClient) GetDetailsForTransaction(txId string) (*ArweavePaymentDetailsResponse, error) {
+	const apiPath = "/arweave/payment/retrieve"
+	client := http.Client{}
+	form := make(url.Values)
+	form.Set("auth_token", apiToken)
+	form.Set("tx_id", txId)
+
+	response, err := client.PostForm(apiURL+apiPath, form)
+	if err != nil {
+		return nil, errors.Err(err)
+	}
+	defer helper.CloseBody(response.Body)
+
+	b, err := io.ReadAll(response.Body)
+	if err != nil {
+		return nil, errors.Err(err)
+	}
+	if response.StatusCode != http.StatusOK {
```

Comment:
Probably want to let through network errors/internal apis hiccups. Only hard fail on real missing tip